WordPress/src/main/java/org/wordpress/android/ui/stories/StoryComposerViewModel.kt

```diff
@@ -82,22 +88,26 @@ class StoryComposerViewModel @Inject constructor(
             editPostRepository.loadPostByLocalPostId(postId.value)
         }
 
-        setupPostEditorAnalyticsSession(postEditorAnalyticsSession)
-
-        notificationType?.let {
-            systemNotificationsTracker.trackTappedNotification(it)
+        // Ensure we have a valid post
+        if (!editPostRepository.hasPost()) {
```

Comment:
Question (❓): What if this logic is moved a bit further up, in between the `notificationType?.let { ... }` and the `if (postId.value == 0) { ... } else { ... }` logic? It seems that the `editPostRepository` is doing some work before checking whether the post is available, like `editPostRepository.loadPostByLocalPostId(...)`, which we might want to avoid. Wdyt?

I am not familiar with the logic and most probably I am wrong, just wanted to mentioned that in case it is something we need to consider.

Reply:
oh! The `EditPostRepository` is ether initialized or loaded with an actual post right before, so it only makes sense to check whether `editPostRepository.hasPost()` after having initialized it :) -  this is actually the earliest spot where we can validate a Post actually exists in the editPostRepository 👍

Reply:
Cool, thanks for letting me know of this Mario! 🌟